### PR TITLE
Add redirect tests as per spec

### LINKED ISSUE #1704: Article redirect tests
**Is your feature request related to a problem? Please describe.**

Where _:service_ is any service that should be served exclusively off .com (that is all non-UK services)

_NB scenarios below in strikethru relating to HTTPS uplift are out of scope till https://github.com/bbc/simorgh-infrastructure/issues/632 is resolved_

**Not .co.uk**

Given I go to https://www.bbc.co.uk/:service/articles/:id
Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id

~**Not http**~

~Given I go to http://www.bbc.co.uk/:service/articles/:id~
~Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id~

**AMP not on .co.uk**

Given I go to https://www.bbc.co.uk/:service/articles/:id.amp
Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id.amp

~**AMP not on http**~

~Given I go to http://www.bbc.co.uk/:service/articles/:id.amp~
~Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id.amp~

**Describe the solution you'd like**
- We should handle redirects within Mozart https://github.com/bbc/simorgh-infrastructure/issues/435. This ticket is only capturing the testing of redirects for our application. 
- These tests should only run against Test and Live environments, not local. 
- These can be run separately to the PR life cycle as a cron job.

Note: that until all of these services are routed on live, the tests should only be run against the test environment. An issue should be created to enable these on live when they are available.

**Describe alternatives you've considered**
We considered having the Cypress end-to-end tests in our current test suite repository, but decided against it - see the discussion thread below

**Testing notes**
We will need to add Cypress tests to make sure that the redirects happen as expected.
`Given I visit URL A, When I am redirected (served a 301 redirect), Then I will land on URL B`

Dev insight: Cypress tests will be sufficient.

### PR DESCRIPTION
**Pull request by jamesbhobbs**
Resolves #1704 

**Overall change:** Tests redirects as per the issue (also tests primary domain)

---

- [x] I have assigned myself to this PR and the corresponding issues
~- [ ] Tests added for new features~
~- [ ] Test engineer approval~

### CODE REVIEW

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]:
```suggestion
    it(`should redirect to *bbc.com/${service}`, () => {
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]: The current values will fail id validation. Try article ids of this format:
```suggestion
        `https://www.bbc.co.uk/${service}/articles/c0000000000o`,
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]:
```suggestion
  describe('WS Redirects', () => {
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]: What's the reason for defining this as a constant rather than just making the line above `Object.keys(services).forEach(service => {`

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:4]:
```suggestion
  const notWSServices = ['news', 'cymrufyw', 'naidheachdan', 'default', 'loadableConfig']; // Not WS
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]: I don't think the `http` links are needed here, since this test is specifically focused on the `.co.uk`->`.com` redirect. 
```suggestion
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:42]: I'd suggest adding a script to the package.json to make code reviewing/testing locally a bit simpler. 
```
    "test:e2e:3rdParty:interactive": "npm run killApp && npm run build && run-p --race start cypress:3rdParty:interactive",
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:31]: If you're going to use JS string methods, I'd suggest this for readability:
```suggestion
          expect(resp.redirectedToUrl.split('/')[2]).to.eq('www.bbc.com')
```
This is because the test is failing since it is trying to match `https://www.bbc.co.u`, cutting off the end of the domain.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (jamesbhobbs) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:4]: This doesn't fail when I run it.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (jamesbhobbs) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:42]: That's a bit too nice, there's already a script, I'm trying to cut the number of scripts.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (jamesbhobbs) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:31]: This is outdated

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:42]: I'll just set up an alias locally on my machine, then.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:4]: Same here. That's interesting. Resolving comment.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:29]: For services arabic, igbo, persian, pidgin, yoruba, the status code is currently a 302, not a 301.

Comment (jamesbhobbs) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:29]: Yes and that will be corrected before this is merged

Comment (jamesdonoh) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:29]: https://github.com/bbc/simorgh/issues/2814

**Comment by jamesdonoh** [comment]
Blocked on https://github.com/bbc/simorgh-infrastructure/issues/633

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (mykeels) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:35]: I'd suggest using `startsWith(...)` and `endsWith` over `substring(...)`.

**Comment by jamesdonoh** [comment]
This is unblocked as https://github.com/bbc/mozart-routing/pull/943 was merged resolving https://github.com/bbc/simorgh-infrastructure/issues/633

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (jamesbhobbs) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:35]: @mykeels feel free to make the change, I don't feel it really matters in an e2e test

**Comment by jamesbhobbs** [comment]
Reassigned to Chris to finish it off

**Comment by chris-hinds** [comment]
Have tested this PR locally and it is working as expected,. @sareh and @mykeels you left some comments here, it looks like they have all be answered/resolved by Benjamin. If you are happy with this PR please can you leave your approval :)

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:18]: We'll be testing these live urls on each environment `local`, `test`, `live`. We should wrap in a conditional `if (Cypress.env('APP_ENV') !== 'local') {}` block, then update these paths to be `https://www.test.bbc.co.uk` on the test env and `https://www.bbc.co.uk` on the live env. Best place for this is to add field `ukBasePath` to `/support/config/envs.js`

**Review by sareh — CHANGES_REQUESTED** [review]

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (mykeels) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:35]: I just re-read the test, and `substring(...)` makes for a better assertion in this case, as you can read the assertion result/error and know immediately what two strings weren't equal.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (chris-hinds) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:18]: Have updated to support this @sareh

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [inline 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]: You should instead be able to do this:

```
import envConfig from '../../support/config/envs';

...
      let urlsTotest = [
        `${envConfig.ukBaseUrl}${service}`,
        ...
```

Then in `envs.js` you can have:

```
{
  live: {
    ukBaseUrl: 'https://www.bbc.co.uk',
    ...
  test: {
    ukBaseUrl: 'https://www.test.bbc.co.uk',
...
  local: {
    ukBaseUrl: 'http://localhost.bbc.co.uk',
...
```

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (chris-hinds) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:18]: @sareh this has been removed now as we do not need to run this tests on different env.

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (chris-hinds) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js]: This change is no longer required

File: 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js
Comment (sareh) [reply 3rdPartyCypress/cypress/integration/worldServiceCoUkRedirect.js:18]: Okay, thanks, Chris.

**Review by sareh — APPROVED** [review]
👍 Looks good. Thanks for the updates.

**Pull request merged**
